assets/services/parachute/parachute.py

```diff
@@ -0,0 +1,209 @@
+
+    # MAV_CMD_SET_MESSAGE_INTERVAL
+    elif msg.command == mavutil.mavlink.MAV_CMD_SET_MESSAGE_INTERVAL:
+        if msg.param1 == mavutil.mavlink.MAVLINK_MSG_ID_PARACHUTE_STATUS:
+            if msg.param2 == -1:
+                parachute_status_thread_event.clear()
+            else:
+                parachute_status_thread_event.set()
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
+            )
+
+    # MAV_CMD_DO_PARACHUTE
+    elif msg.command == mavutil.mavlink.MAV_CMD_DO_PARACHUTE:
+        if msg.param1 == mavutil.mavlink.PARACHUTE_RELEASE:
+            if (
+                parachute_status[ParachuteStatusFields.ARM_STATUS.value]
+                & mavutil.mavlink.PARACHUTE_TRIGGER_FLAGS_FC
+            ):
+                if (
+                    parachute_status[ParachuteStatusFields.SAFETY_STATUS.value]
+                    & mavutil.mavlink.PARACHUTE_SAFETY_FLAGS_GROUND_CLEARED
+                ):
+                    mavlink_connection_out.mav.command_ack_send(
+                        msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+                    )
+                    update_parachute_status(
+                        ParachuteStatusFields.DEPLOYMENT_STATUS,
+                        int(mavutil.mavlink.PARACHUTE_DEPLOYMENT_TRIGGER_DRONE),
+                    )
+                else:
+                    mavlink_connection_out.mav.command_ack_send(
+                        msg.command, result=mavutil.mavlink.MAV_RESULT_TEMPORARILY_REJECTED
+                    )
+            else:
+                mavlink_connection_out.mav.command_ack_send(
+                    msg.command, result=mavutil.mavlink.MAV_RESULT_FAILED
+                )
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
+            )
+
+    # MAV_CMD_SET_PARACHUTE_ARM
+    elif msg.command == mavutil.mavlink.MAV_CMD_SET_PARACHUTE_ARM:
+        arm_flags = int(msg.param1)
+        bitmask = int(msg.param2)
+        parachute_trigger_source_invalid = (
+            (mavutil.mavlink.PARACHUTE_TRIGGER_FLAGS_ENUM_END - 1) << 1
+        ) - 1
+        if (
+            arm_flags & ~parachute_trigger_source_invalid
+            or bitmask & ~parachute_trigger_source_invalid
+        ):
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
+            )
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+            flag = 1
+            new_arm_status = 0
+            while flag < mavutil.mavlink.PARACHUTE_TRIGGER_FLAGS_ENUM_END:
```

Comment:
I think you can do this in 2 operations instead of a loop:
1. Start with original flags
2. Clear all `bitmask` bits (`new = original & ~bitmask`)
3. Set all `arm_flag` bits that are also set in `bitmask` (`new &= arm_flags & bitmask`)

Reply:
for step 3, did you mean `new |= arm_flags & bitmask`?